istioctl/pkg/validate/validate.go

```diff
@@ -171,16 +280,24 @@ func NewValidateCommand() *cobra.Command {
 	var referential bool
 
 	c := &cobra.Command{
-		Use:     "validate -f FILENAME [options]",
-		Short:   "Validate Istio policy and rules",
-		Example: `istioctl validate -f bookinfo-gateway.yaml`,
-		Args:    cobra.NoArgs,
+		Use:   "validate -f FILENAME [options]",
+		Short: "Validate Istio policy and rules",
+		Example: `
+		# Validate bookinfo-gateway.yaml
+		istioctl validate -f bookinfo-gateway.yaml
+		
+		# Validate current deployments within the cluster
+		kubectl get deployments -o yaml |istioctl validate -f -
+`,
+		Args: cobra.NoArgs,
 		RunE: func(c *cobra.Command, _ []string) error {
-			return validateFiles(filenames, referential)
+			return validateFiles(filenames, referential, c.OutOrStderr())
 		},
 	}
 
 	flags := c.PersistentFlags()
+	flags.StringVarP(&istioNamespace, "istioNamespace", "i", kube.IstioNamespace,
```

Comment:
I don't know how I feel about this flag.  This API change would be permanent and we are heading into a world where there could be more than 1 istio namespace.  Perhaps the answer is to punt on the UX part, and hardcode `istio-system` as the expected istioNamespace.

Reply:
Another option is to treat it as a comma separated list and call it plural `istioNamespaces`.

Reply:
This command will work against one of your Istio installation, so if you have multiple Istio installation within the cluster, you can use ```-i``` option to point to the istio you want to validate against, something like ``` istioctl validate -f bookinfo-gateway.yaml -i test```

Reply:
There is already a `istioNamespace` in _istioctl/cmd/root.go_.  If you do `istioctl validate --help` you will see it is already there.  Please re-use that variable instead of creating a duplicate.

Reply:
Done